src/main/java/org/javarosa/core/model/instance/ExternalDataInstance.java

```diff
@@ -68,10 +68,12 @@ public static ExternalDataInstance build(String instanceSrc, String instanceId)
         return new ExternalDataInstance(root, instanceId, instanceSrc);
     }
 
-    private static TreeElement parseExternalInstance(String instanceSrc, String instanceId) throws IOException, InvalidReferenceException, InvalidStructureException, XmlPullParserException, UnfullfilledRequirementsException {
+    private static TreeElement parseExternalInstance(String instanceSrc, String instanceId)
+        throws IOException, InvalidReferenceException, InvalidStructureException, XmlPullParserException, UnfullfilledRequirementsException {
         String path = getPath(instanceSrc);
-        return instanceSrc.contains("file-csv") ?
-            CsvExternalInstance.parse(instanceId, path) : XmlExternalInstance.parse(instanceId, path);
+        return instanceSrc.contains("file-csv") ? CsvExternalInstance.parse(instanceId, path)
```

Comment:
I considered pulling this out into a more conventional factory so it could be tested but I don't see a big benefit given how everything else is structured at the moment.